parser/parsetypes/conn.go

```diff
@@ -65,3 +64,8 @@ func (in *Conn) TargetCollection(config *config.StructureTableCfg) string {
 func (in *Conn) Indices() []string {
 	return []string{"$hashed:id_orig_h", "$hashed:id_resp_h", "-duration", "ts", "uid"}
 }
+
+//ConvertFromJSON performs any extra conversions necessary when reading from JSON
+func (in *Conn) ConvertFromJSON() {
+	in.TimeStamp = int64(in.TimeStampFloat)
+}
```

Comment:
The remaining piece is this function, which is called after unmarshalling the json. This takes the float field that `ts` was read into and typecasts it to the original `TimeStamp` field. If we have more issues like this in the future we can add more conversion steps to this function.

Reply:
I wish we didn't have to include this function with the same definition in each parse type. It would be nice to have one method that could handle this for all of the types. We'd either have to use reflection or make sure to maintain a large typeswitch to accomplish that though. What are your thoughts?

Reply:
I wish it wasn't like this either. I explored some different options before settling on this one.

I'm not a huge fan of reflection and my impression is that it reduces performance. Though we already use it for the Bro log parsing and I think the json library uses it too. If we could get it to work I don't think it would be too convoluted or noticeably impact performance. And the `convertTimestamp` function already uses a type switch which I could argue as being a type of reflection.

It's not often we have to add a new type and my preference would be to leave it how it is. But be open to switching if we ever refactor the parser in the future.

Reply:
Agreed... Currently the affect on import time seems to be negligible, even with large datasets. This is the most consistent option we've found so far... if we refactor later maybe a new one will be available, but for now this is what works.